### PR TITLE
Add Cloud provider disclaimer

### PR DESCRIPTION
**Pull request by PenghaiZhang**

### CODE REVIEW

File: Source/Plugins/Core/com.equella.core/js/tsrc/cloudprovider/CloudProviderDisclaimerDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  render() {
+    const { openDialog, onClose } = this.props;
+    return (
+      <Dialog fullWidth open={openDialog} onClose={onClose}>
+        <DialogTitle>
+          {cloudProviderLangStrings.newcloudprovider.disclaimer.title}
+          <IconButton
+            aria-label="Close"
+            onClick={onClose}
+            style={{ position: "absolute", right: "10px" }}
```
Comment (SammyIsConfused) [inline Source/Plugins/Core/com.equella.core/js/tsrc/cloudprovider/CloudProviderDisclaimerDialog.tsx]: Inline styles like this are an anti pattern in react. Best to either use a material UI theme, or at least save these styles as a constant of type CSSProperties.

File: Source/Plugins/Core/com.equella.core/js/tsrc/cloudprovider/CloudProviderDisclaimerDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            <CloseIcon />
+          </IconButton>
+        </DialogTitle>
+        <DialogContent>
+          <Typography align="justify">
+            You acknowledge that the software, products and services{" "}
+            <b>(Materials)</b> made available through the Cloud Provider Portal
+            are provided to you by the relevant Cloud Provider and have not been
+            authenticated or endorsed in whole or in part by any third party.
+            Your use of the Materials is governed solely by the terms and
```
Comment (SammyIsConfused) [inline Source/Plugins/Core/com.equella.core/js/tsrc/cloudprovider/CloudProviderDisclaimerDialog.tsx]: I'd definitely consider saving this long string as  a constant, or even to it's own text file.

File: Source/Plugins/Core/com.equella.core/js/tsrc/cloudprovider/CloudProviderAddDialog.tsx [hunk 3]
```diff
@@ -79,6 +96,15 @@ export default class CloudProviderAddDialog extends React.Component<
               error={!isUrlValid}
               helperText={cloudProviderLangStrings.newcloudprovider.help}
             />
+
+            <Typography variant="body2" style={{ marginTop: "10px" }}>
```
Comment (SammyIsConfused) [inline Source/Plugins/Core/com.equella.core/js/tsrc/cloudprovider/CloudProviderAddDialog.tsx]: see below comment regarding inline styling.

File: Source/Plugins/Core/com.equella.core/js/tsrc/cloudprovider/CloudProviderDisclaimerDialog.tsx (hunk 2 above)
Comment (edalex-ian) [reply Source/Plugins/Core/com.equella.core/js/tsrc/cloudprovider/CloudProviderDisclaimerDialog.tsx]: I'm okay with this one, as it's kinda exceptional. It's not worth the additional complexity - in a general sense.

File: Source/Plugins/Core/com.equella.core/js/tsrc/cloudprovider/CloudProviderDisclaimerDialog.tsx (hunk 1 above)
Comment (edalex-ian) [reply Source/Plugins/Core/com.equella.core/js/tsrc/cloudprovider/CloudProviderDisclaimerDialog.tsx]: Yup, good to sort that one out.

**Pull request merged**